client/state/http/actions.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Internal dependencies
+ */
+import { HTTP_REQUEST } from 'state/action-types';
+
+/***
+ * @typedef {Object} RequestDescription
+ * @property {String} url the url to request
+ * @property {String} method the method we should use in the request: GET, POST etc.
+ * @property {Array<Array<String>>} headers array of [ 'key', 'value' ] pairs for the request headers
+ * @property {Object} body data send as body
+ * @property {Boolean} withCredentials save cookie set on request
```

Comment:
It's the "standard" name for it, I rather not rename it, that's what `XMLHttpRequest` and `fetch`.

https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/withCredentials

`fetch` has a slightly different semantics for that, but still calls it credentials:
https://developer.mozilla.org/en-US/docs/Web/API/Request/credentials